scripts/validate_rst_title_capitalization.py

```diff
@@ -0,0 +1,306 @@
+#!/usr/bin/env python
+
+"""
+GH #29641
+
+Collect the titles in the rst files and validate if they follow the proper
+capitalization convention.
+
+Prints the titles that do not follow the convention.
+
+Usage::
+./scripts/validate_rst_title_capitalization.py doc/source/development/contributing.rst
+./scripts/validate_rst_title_capitalization.py doc/source/
+
+"""
+
+import argparse
+import sys
+import re
+import os
+from os import walk
+from typing import Generator, List
+
+
+# Keynames that would not follow capitalization convention
+CAPITALIZATION_EXCEPTIONS = {
+    "pandas",
+    "Python",
+    "IPython",
+    "PyTables",
+    "Excel",
+    "JSON",
+    "HTML",
+    "SAS",
+    "SQL",
+    "BigQuery",
+    "STATA",
+    "Interval",
+    "PEP8",
+    "Period",
+    "Series",
+    "Index",
+    "DataFrame",
+    "C",
+    "Git",
+    "GitHub",
+    "NumPy",
+    "Apache",
+    "Arrow",
+    "Parquet",
+    "MultiIndex",
+    "NumFOCUS",
+    "sklearn",
+    "Docker",
+}
+
+# Lowercase representation of CAPITALIZATION_EXCEPTIONS
+CAPITALIZATION_EXCEPTIONS_LOWER = {word.lower() for word in CAPITALIZATION_EXCEPTIONS}
+
+# Dictionary of bad titles that will be printed later along with line numbers
+# Key: Document Directory, Value: Pair(Bad Title, Line Number)
+bad_title_dict = {}
+
+# Error Message:
+err_msg = "Heading capitalization formatted incorrectly. Please correctly capitalize"
+
+
+def follow_capitalization_convention(title: str) -> bool:
```

Comment:
Since this is returning a bool, I think that ```is_following_capitalization_convention``` is more appropriate function name.